code/modules/projectiles/projectile.dm

```diff
@@ -258,6 +276,17 @@
 		error = dist_x/2 - dist_y
 	else
 		error = dist_y/2 - dist_x
+
+	target_angle = round(Get_Angle(starting,target))
+
+	if(linear_movement)
+		//If the icon has not been added yet
+		if( !("[icon_state][target_angle]" in bullet_master) )
+			var/icon/I = new(icon,"[icon_state]_pixel") //Generate it.
+			I.Turn(target_angle+45)

```

Comment:
Doubt it'd change anything, the issue is that using matrix transform here cause the sprite to always face a cardinal direction (the projectile sprites are diagonal)